packages/hub/src/session.ts

```diff
@@ -123,3 +134,120 @@ export class Session implements SessionInterface {
     });
   }
 }
+
+type releaseHealthAttributes = {
+  environment?: string;
+  release: string;
+};
+
+/**
+ * @inheritdoc
+ */
+export class SessionFlusher implements SessionFlusherInterface {
+  public readonly flushTimeout: number = 60;
+  private _pendingAggregates: { [key: number]: AggregationCounts } = {};
+  private _sessionAttrs: releaseHealthAttributes;
+  private _intervalId: ReturnType<typeof setInterval>;
+  private _isEnabled: boolean = true;
+  private _transport: Transport;
+
+  constructor(transport: Transport, attrs: releaseHealthAttributes) {
+    this._transport = transport;
+    // Call to setInterval, so that flush is called every 60 seconds
+    this._intervalId = setInterval(() => this.flush(), this.flushTimeout * 1000);
+    this._sessionAttrs = attrs;
+  }
+
+  /** Sends session aggregates to Transport */
+  public sendSessionAggregates(sessionAggregates: SessionAggregates): void {
+    if (!this._transport.sendSessionAggregates) {
+      logger.warn("Dropping session because custom transport doesn't implement sendSessionAggregates");
+      return;
+    }
+    this._transport.sendSessionAggregates(sessionAggregates).then(null, reason => {
+      logger.error(`Error while sending session: ${reason}`);
+    });
+  }
+
+  /** Checks if `pendingAggregates` has entries, and if it does flushes them by calling `sendSessions` */
+  flush(): void {
+    const sessionAggregates = this.getSessionAggregates();
+    if (sessionAggregates.aggregates.length === 0) {
+      return;
+    }
+    this._pendingAggregates = {};
+    this.sendSessionAggregates(sessionAggregates);
+  }
+
+  /** Massages the entries in `pendingAggregates` and returns aggregated sessions */
+  getSessionAggregates(): SessionAggregates {
+    const aggregates: AggregationCounts[] = Object.keys(this._pendingAggregates).map((key: string) => {
+      return this._pendingAggregates[parseInt(key)];
+    });
+
+    const sessionAggregates: SessionAggregates = {
+      attrs: this._sessionAttrs,
+      aggregates: aggregates,
+    };
+    return dropUndefinedKeys(sessionAggregates);
+  }
+
+  /** JSDoc */
+  close(): void {
+    clearInterval(this._intervalId);
+    this._isEnabled = false;
+    this.flush();
+  }
+
+  /**
+   * Wrapper function for _incrementSessionStatusCount that checks if the instance of SessionFlusher is enabled then
+   * fetches the session status of the request from `_requestSession.status` on the scope and passes them to
+   * `_incrementSessionStatusCount` along with the start date
+   */
+  public incrementSessionStatusCount(): void {
+    if (!this._isEnabled) {
+      return;
+    }
+    const scope = getCurrentHub().getScope();
+    /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+    const requestSession = (scope as any)._requestSession;
```

Comment:
All these `typescript-eslint/no-unsafe-member-access` would be gone as well then.

Reply:
Even though I do agree with you, why would we want to increase the surface area of the Scope API to add a getter function that the user should never access anyways?

Reply:
User very rarely interact directly with the scope. Eg. `getUser` is available on scope, but not on `Hub`, so the "global" API is not extended. If we won't do this, then we need to come up with other idea. Otherwise using typescript is basically pointless.

Reply:
If not getter, then make the field public, or write `get requestSession() { ... }` method to make a property getter.